engine/apps/alerts/tasks/notify_user.py

```diff
@@ -172,16 +215,50 @@ def notify_user_task(
                         notification_error_code=UserNotificationPolicyLogRecord.ERROR_NOTIFICATION_POSTING_TO_SLACK_IS_DISABLED,
                     )
                 else:
-                    log_record = UserNotificationPolicyLogRecord(
-                        author=user,
-                        type=UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_TRIGGERED,
-                        notification_policy=notification_policy,
-                        alert_group=alert_group,
-                        reason=reason,
-                        slack_prevent_posting=prevent_posting_to_thread,
-                        notification_step=notification_policy.step,
-                        notification_channel=notification_policy.notify_by,
-                    )
+                    if (
+                        settings.FEATURE_NOTIFICATION_BUNDLE_ENABLED
+                        and UserNotificationBundle.notification_is_bundleable(notification_policy.notify_by)
+                    ):
+                        user_notification_bundle, _ = UserNotificationBundle.objects.select_for_update().get_or_create(
```

Comment:
why `select_for_update()` and then `get_or_create()`?

Reply:
it works the same way as
```
user_notification_bundle, _ = UserNotificationBundle.objects.get_or_create(
    user=user, important=important, notification_channel=notification_policy.notify_by
)
user_notification_bundle = UserNotificationBundle.objects.filter(pk=user_notification_bundle.pk).select_for_update()[0]
```